src/sync/installation.ts

```diff
@@ -236,215 +233,201 @@ export const isNotFoundError = (
+		const jiraClient = await getJiraClient(
+			subscription.jiraHost,
+			installationId,
+			logger
+		);
+		const github = await getEnhancedGitHub(app, installationId);
+		const nextTask = await getNextTask(subscription);
+
+		if (!nextTask) {
+			await subscription.update({ syncStatus: "COMPLETE" });
+			statsd.increment(metricSyncStatus.complete);
+			logger.info({ job, task: nextTask }, "Sync complete");
+			return;
+		}
 
-			if (!nextTask) {
-				await subscription.update({ syncStatus: "COMPLETE" });
-				statsd.increment(metricSyncStatus.complete);
-				logger.info({ job, task: nextTask }, "Sync complete");
-				return;
-			}
+		await subscription.update({ syncStatus: "ACTIVE" });
 
-			await subscription.update({ syncStatus: "ACTIVE" });
-
-			const { task, repositoryId, cursor } = nextTask;
-			let { repository } = nextTask;
-
-			if (!repository) {
-				// Old records don't have this info. New ones have it
-				const { data: repo } = await github.request("GET /repositories/:id", {
-					id: repositoryId
-				});
-				repository = getRepositorySummary(repo);
-				await subscription.updateSyncState({
-					repos: {
-						[repository.id]: {
-							repository
-						}
-					}
-				});
-			}
+		const { task, repositoryId, cursor } = nextTask;
+		let { repository } = nextTask;
 
-			logger.info({ job, task: nextTask }, "Starting task");
-
-			const processor = tasks[task];
-
-			const execute = async () => {
-				if (await booleanFlag(BooleanFlags.SIMPLER_PROCESSOR, true)) {
-
-					// just try with one page size
-					return await processor(github, repository, cursor, 20);
-
-				} else {
-
-					for (const perPage of [20, 10, 5, 1]) {
-						// try for decreasing page sizes in case GitHub returns errors that should be retryable with smaller requests
-						try {
-							return await processor(github, repository, cursor, perPage);
-						} catch (err) {
-							logger.error({
-								err,
-								job,
-								github,
-								repository,
-								cursor,
-								task
-							}, `Error processing job with page size ${perPage}, retrying with next smallest page size`);
-							if (isRetryableWithSmallerRequest(err)) {
-								// error is retryable, retrying with next smaller page size
-								continue;
-							} else {
-								// error is not retryable, re-throwing it
-								throw err;
-							}
-						}
+		if (!repository) {
+			// Old records don't have this info. New ones have it
+			const { data: repo } = await github.request("GET /repositories/:id", {
+				id: repositoryId
+			});
+			repository = getRepositorySummary(repo);
+			await subscription.updateSyncState({
+				repos: {
+					[repository.id]: {
+						repository
 					}
-
 				}
+			});
+		}
+
+		logger.info({ job, task: nextTask }, "Starting task");
+
+		const processor = tasks[task];
 
-				throw new Error(`Error processing GraphQL query: installationId=${installationId}, repositoryId=${repositoryId}, task=${task}`);
-			};
+		const handleGitHubError = (err) => {
```

Comment:
I removed this function in an earlier PR, so this must be a fragment from a merge conflict. I replaced this function with `isRetryableWithSmallerRequest()`, which returns true or false depending on whether we think it's viable to retry the request with a smaller page size.

This needs to be used in conjunction with the for loop that goes through the different page sizes. Please compare with the current `master` again.

Reply:
I don't think we need this functionality at all if we code the graphql API correctly with good error handling.